menu/menu.cpp

```diff
@@ -152,31 +134,93 @@ void menu::menu::subMenu() {
         case 4:
             std::cout << Term::move_cursor(17, 7)
                       << Term::color24_fg(255, 64, 0) << "    Press Q";
-            arrow = false;
+            settingsSub = true;
             anim::snakeSettings();
             break;
         case 5:
             std::cout << Term::move_cursor(17, 7)
                       << Term::color24_fg(255, 64, 0) << "    Press Q";
-            arrow = false;
+            selector = false;
             anim::snakeAbout();
             break;
         case 6:
             anim::snakeLicense();
-            arrow = false;
-            license_sub_open = true;
-            license_page_toggle = true;
-            anim::license_menu_toggle(license_page_toggle);
+            selector = false;
+            licenseSub = true;
+            licensePageToggle = true;
+            anim::license_menu_toggle(licensePageToggle);
             break;
     }
-    lastCursorState = cursorState;
+    if (!settingsSubSub)
+        lastCursorState = cursorState;
+    cursorState = 0;
+    updateDescription();
+    std::cout << Term::color(Term::fg::reset) << std::flush;
+}
+void menu::menu::handleGamemode() {
+    sub = false;
+    gamemodeSub = false;
+    draw::clearField();
+    snake::snake snakeObj;
+
+    switch (cursorState) {
+        case 0:  // SNEK
+            snakeObj.run();
+            break;
+        case 1:  // Classic
+            snakeObj.run();
+            break;
+        case 2:  // Infinite
+            snakeObj.run();
+            break;
+        case 3:  // Chaos
+            snakeObj.run();
+            break;
+        case 4:  // Creative
+            snakeObj.run();
+            break;
+    }
+
+    draw::snakeMenuBase();
     cursorState = 0;
     updateDescription();
+    std::cout << Term::move_cursor(9, 20) << Term::color24_fg(255, 64, 0)
+              << "<--" << Term::color(Term::fg::reset) << std::flush;
+}
+void menu::menu::handleSettings() {
+    settingsSub = false;
+    settingsSubSub = true;
+    draw::clearMenu();
+
+    switch (cursorState) {
+        case 0:
+            std::cout << Term::move_cursor(7, 10)
+                      << Term::color24_fg(255, 255, 0) << "CONTROLS!";
+            // updateDescription();
+            // anim::snakeControls();
+            break;
+        case 1:
+            std::cout << Term::move_cursor(7, 11)
+                      << Term::color24_fg(255, 255, 0) << "THEMES";
+            // updateDescription();
+            // anim::snakeThemes();
+            break;
+        case 2:
+            std::cout << Term::move_cursor(7, 9)
+                      << Term::color24_fg(255, 255, 0) << "PLACEHOLDER";
+            // updateDescription();
+            // anim::snakePlaceholder();
+            break;
+    }
+    cursorState = 0;
+    updateCursor(cursor::notMoved);
+    // updateDescription();
     std::cout << Term::color(Term::fg::reset) << std::flush;
 }
+// TODO: void handleControls() {}
```

Comment:
Is this important?

Reply:
Yes, this will be the function for opening/interacting with the controls sub-menu.